docs/dev/developing-secret-contracts.md

```diff
@@ -0,0 +1,157 @@
+# Developing Secret Contracts
+
+Secret Contacts are based on CosmWasm v0.10.
+
+Check out the [CosmWasm docs](https://docs.cosmwasm.com) as well. They are probably more extensive.
+
+Don't forget to go over the [differences between SecretWasm and CosmWasm](#differences-from-cosmwasm).
+
+- [Developing Secret Contracts](#Developing-Secret-Contracts)
+  - [IDEs](#IDEs)
+  - [Init](#Init)
+  - [Handle](#Handle)
+  - [Query](#Query)
+  - [Inputs](#Inputs)
+  - [APIs](#APIs)
+  - [State](#State)
+  - [Some libraries/crates considerations](#Some-librariescrates-considerations)
+  - [Randomness](#Randomness)
+    - [Roll your own](#Roll-your-own)
+      - [Poker deck shuffling example](#Poker-deck-shuffling-example)
+    - [Use an external oracle](#Use-an-external-oracle)
+  - [Outputs](#Outputs)
+  - [External query](#External-query)
+  - [Compiling](#Compiling)
+  - [Storing and deploying](#Storing-and-deploying)
+  - [Verifying your contract code on explorers](#Verifying-your-contract-code-on-explorers)
+  - [Testing](#Testing)
+  - [Debugging](#Debugging)
+- [Differences from CosmWasm](#Differences-from-CosmWasm)
+
+## IDEs
+
+Secret Contracts are developed with the [Rust](https://www.rust-lang.org/) programming language and compiled to [WASM](https://webassembly.org/) binaries.
+
+These IDEs are known to work very well for developing Secret Contracts:
+
+- [CLion](https://www.jetbrains.com/clion/)
+- [VSCode](https://code.visualstudio.com/) with the [rust-analyzer](https://rust-analyzer.github.io/) extension
+
+## Init
+
+`init` is the constructor of your contract. This function is called only once in the lifetime of the contract.
+
+Example Invocation from `secretcli`:
+
+```bash
+secretcli tx compute instantiate "$CODE_ID" "$INPUT_MSG" --label "$UNIQUE_LABEL" --from "$MY_KEY"
+```
+
+Example Invocation from `SecretJS`:
+
+```js
+// TODO
```

Comment:
Before you can init here you'll need funded account and a client connection, I don't know if that's worth a mention.
All the snippets in the TODOs can come from [here](https://github.com/enigmampc/enigma-blockchain-contracts-guide/blob/testnet-quickstart/cosmwasm-js.md)

I'd suggest you try it too, but currently the cli won't work unless you [install it locally from here](https://github.com/enigmampc/SecretNetwork/pull/481/files). That said the code and config works as it is outside the cli, eg holdem, voting app etc

Reply:
You're right. This is just supposed to be a context-less code snippet, to show to devs that it's really easy to do.